zebrad/tests/acceptance.rs

```diff
@@ -461,3 +461,70 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn metrics_tracing_listeners() -> Result<()> {
+    use hyper::{Client, Uri};
+
+    zebra_test::init();
+
+    // From here we dont know what port the OS selected for us if we use port 0 so we use a hard coded port for the 2 endpoints.
+    let metrics_endpoint = "127.0.0.1:9998";
+    let metrics_url = "http://127.0.0.1:9998";
+    let tracing_endpoint = "127.0.0.1:9999";
+    let tracing_url = "http://127.0.0.1:9999";
+
+    // Write a configuration that has both metrics endpoint_addr and tracing endpoint_addr options set
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(metrics_endpoint.parse().unwrap());
+    config.tracing.endpoint_addr = Some(tracing_endpoint.parse().unwrap());
+
+    let dir = TempDir::new("zebrad_tests")?;
+    fs::File::create(dir.path().join("zebrad.toml"))?
+        .write_all(toml::to_string(&config)?.as_bytes())?;
+
+    let tempdir = dir.path().to_path_buf();
+
+    let mut child = get_child(&["start"], &tempdir)?;
+
+    // Run the program and kill it at 1 second
+    std::thread::sleep(Duration::from_secs(1));
+
+    // Test metrics endpoint
+    let metrics_client = Client::new();
+    let metrics_res = metrics_client.get(Uri::from_static(metrics_url)).await?;
+    assert!(metrics_res.status().is_success());
+    let metrics_body = hyper::body::to_bytes(metrics_res).await?;
+    assert!(std::str::from_utf8(&metrics_body)
+        .unwrap()
+        .contains("metrics snapshot"));
+
+    // Test tracing endpoint
+    let tracing_client = Client::new();
```

Comment:
do we need to create a second `Client`?

Reply:
No need, changed to 1 client as part of https://github.com/ZcashFoundation/zebra/pull/1000/commits/939349cb9413391f2564a787246f8a1e0c96e3dc